app/src/main/java/com/neiapp/spocan/ui/activity/LoginActivity.java

```diff
@@ -135,17 +138,49 @@ public void onStart() {
 
     private void updateUI(FirebaseUser firebaseUser) {
         mSignOutBtn.setVisibility(View.VISIBLE);
-        firebaseUser.getIdToken(true).addOnCompleteListener(task ->{
+        firebaseUser.getIdToken(true).addOnCompleteListener(task -> {
             GetTokenResult result = task.getResult();
             String token = result.getToken();
-
-            // Todo llamar a Backend.authenticate(token, CallbackInstance<User>)
             // Ese método va a autenticar el token de firebase contra el backend y guardar el jwt estático en RestClientBackend,
-            // finalmente va a devolver el User. Si no es nulo redirigir al main, si es nulo redirigir al registro.
+            // finalmente va a devolver el User.
+            Backend.authenticate(token, new CallbackInstance<User>() {
+                @Override
+                public void onFailure(String message, Integer httpStatus) {
+                    int errorCount = 0;
+                    if (httpStatus != null) {
+                        if (httpStatus == HTTPCodes.NOT_ACCEPTABLE.getCode() || httpStatus == HTTPCodes.BAD_REQUEST_ERROR.getCode()) {
+                            Toast.makeText(getApplicationContext(), "Token invalido o no autorizado", Toast.LENGTH_LONG).show();
+                            errorCount++;
+                        } else if (httpStatus == HTTPCodes.SERVER_ERROR.getCode()) {
+                            Toast.makeText(getApplicationContext(), "Error del servidor, intente de nuevo mas tarde", Toast.LENGTH_LONG).show();
+                            errorCount++;
+                        } else {
+                            Toast.makeText(getApplicationContext(), "Error desconocido", Toast.LENGTH_LONG).show();
+                        }
+
+
+                    }
+
+                }
+
+                @Override
+                public void onSuccess(User instance) {
+                    // TODO: (SPOCAN- 32) Si no es nulo redirigir al main, si es nulo redirigir al registro.
+                    final Intent intent;
+                    if (instance != null) {
+                        intent = new Intent(getApplicationContext(), MainActivity.class);
+                        intent.putExtra("Instancia BE", instance);
```

Comment:
Podemos incluir una constante pública del lado del MainActivity que se llame USER en reemplazo de "Instancia BE"?

Reply:
donde pusiste MainActivity quisiste decir LoginActivity? O creo esa constante en main y de alguna forma la llamo ahi en el tag ese?

Reply:
A mi me gusta más que la key esté del lado de quien la va a usar, pero si o si tiene que ser pública para que desde el Login la puedas acceder. Igual mandala donde a vos te parezca!